### PR TITLE
WIP: Integrate Searx metasearch engine into FreedomBox

### PR DESCRIPTION
**Pull request by JosephKiranBabu**
The Debian package of Searx is not in a working condition yet. There's a patch that must be merged.
This PR is functionally complete but I'm not totally confident about it since I haven't seen it working with the current Searx Debian package.

Please do not risk merging this in release 0.21.0

### CODE REVIEW

File: actions/searx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+def subcommand_setup(_):
+    """Post installation actions for Searx"""
+    _copy_uwsgi_configuration()
+    _generate_secret_key()
+
+
+def subcommand_enable(_):
+    """Enable web configuration and reload."""
+    action_utils.webserver_enable('searx-plinth')
```
Comment (SunilMohanAdapa) [inline actions/searx]: Should be consider enabling/disabling uwsgi application configuration?

File: plinth/utils.py [hunk 2]
```diff
@@ -136,3 +137,21 @@ def grep(pattern, file_name):
...
+    mode: an octal number to specify file permissions
+    """
+    output_dir = os.path.dirname(output_file)
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir, mode=0o755)
+
+    with gzip.open(gzip_file, 'rb') as f:
+        contents = f.read()
+    with open(output_file, 'wb') as f:
+        f.write(contents)
```
Comment (SunilMohanAdapa) [inline plinth/utils.py]: Please use `file_handle` (or so) instead of `f` for consistent naming of variables.

File: actions/searx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    settings_file = '/etc/searx/settings.yml'
+
+    # Create settings file if not exists
+    if not os.path.exists(settings_file):
+        example_settings_file = '/usr/share/doc/searx/examples/settings.yml.gz'
+        gunzip(example_settings_file, settings_file)
+
+    # Generate and set a secret key
+    with YAMLFile(settings_file) as settings:
+        secret_key = os.urandom(16).hex()
```
Comment (SunilMohanAdapa) [inline actions/searx]: Searx is using this as a key for doing HMAC-SHA256. A minimum key size of 32 bytes is recommended. 64 bytes is probably better.

Using the `secrets` module might be a better than using os.urandom() directly.

**Review by SunilMohanAdapa — CHANGES_REQUESTED** [review]
Here are some preliminary comments.

**Comment by jvalleroy** [comment]
Repository moved to https://salsa.debian.org/freedombox-team/freedombox.

**Pull request closed without merging**